apps/studio/components/interfaces/Storage/StorageSettings/StorageFileSizeLimitErrorMessage.tsx

```diff
@@ -0,0 +1,105 @@
+import Link from 'next/link'
+import { type FieldError } from 'react-hook-form'
+
+import { InlineLink } from 'components/ui/InlineLink'
+import { LARGEST_SIZE_LIMIT_BUCKETS_COUNT } from 'data/sql/queries/get-largest-size-limit-buckets'
+import { cn, Tooltip, TooltipContent, TooltipTrigger } from 'ui'
+import {
+  decodeBucketLimitErrorMessage,
+  formatBytesForDisplay,
+  isBucketLimitErrorMessage,
+} from './StorageSettings.utils'
+
+interface StorageFileSizeLimitErrorMessageProps {
+  error?: FieldError
+  projectRef?: string
+}
+
+export const StorageFileSizeLimitErrorMessage = ({
+  error,
+  projectRef,
+}: StorageFileSizeLimitErrorMessageProps) => {
+  if (!error?.message) return null
+
+  const isBucketLimitError = isBucketLimitErrorMessage(error.message)
+  if (!isBucketLimitError) return <>{error.message}</>
+
+  const bucketLimitErrorBuckets = decodeBucketLimitErrorMessage(error.message)
+  if (bucketLimitErrorBuckets.length === 0) return null
+
+  const hasUnfetchedBuckets = bucketLimitErrorBuckets.length > LARGEST_SIZE_LIMIT_BUCKETS_COUNT
+
+  const [primaryBucket, ...remainingBuckets] = bucketLimitErrorBuckets
+  const otherBuckets = remainingBuckets.slice(0, LARGEST_SIZE_LIMIT_BUCKETS_COUNT - 1)
+
+  const primaryBucketLimit = formatBytesForDisplay(primaryBucket.limit)
+
+  const primaryBucketLink =
+    projectRef !== undefined ? (
+      <InlineLink
+        href={`/project/${projectRef}/storage/files/buckets/${primaryBucket.name}`}
```

Comment:
could link to open the edit modal directly

```suggestion
        href={`/project/${projectRef}/storage/files/buckets/${primaryBucket.name}?edit=true`}
```